src/main/java/edu/cornell/kfs/fp/businessobject/ScheduledSourceAccountingLine.java

```diff
@@ -0,0 +1,113 @@
+	}
+
+	public KualiDecimal getPartialAmount() {
+		return partialAmount;
+	}
+
+	public void setPartialAmount(KualiDecimal partialAmount) {
+		this.partialAmount = partialAmount;
+	}
+
+	public Date getEndDate() {
+		return endDate;
+	}
+
+	public void setEndDate(Date endDate) {
+		this.endDate = endDate;
+	}
+
+	public Date getStartDate() {
+		return startDate;
+	}
+
+	public void setStartDate(Date startDate) {
+		this.startDate = startDate;
+	}
+
+	public String getScheduleType() {
+		return scheduleType;
+	}
+
+	public void setScheduleType(String scheduleType) {
+		this.scheduleType = scheduleType;
+	}
+
+	@Override
+	public Map getValuesMap() {
+		Map simpleValues = super.getValuesMap();
+		simpleValues.put(CuFPConstants.ScheduledSourceAccountingLineConstants.PARTIAL_TRANSACTION_COUNT, partialTransactionCount);
+		simpleValues.put(CuFPConstants.ScheduledSourceAccountingLineConstants.PARTIAL_AMOUNT, partialAmount);
+		simpleValues.put(CuFPConstants.ScheduledSourceAccountingLineConstants.START_DATE, startDate);
+		simpleValues.put(CuFPConstants.ScheduledSourceAccountingLineConstants.END_DATE, endDate);
+		simpleValues.put(CuFPConstants.ScheduledSourceAccountingLineConstants.SCHEDULE_TYPE, scheduleType);
+		return simpleValues;
+	}
+
+	@Override
+	public boolean isSourceAccountingLine() {
+		return true;
+	}
+
+	@Override
+	public boolean isTargetAccountingLine() {
+		return false;
+	}
+	
+	@Override
+	public void copyFrom(AccountingLine other) {
+		super.copyFrom(other);
+		if (ScheduledSourceAccountingLine.class.isAssignableFrom(other.getClass())) {
+			ScheduledSourceAccountingLine fromLine = (ScheduledSourceAccountingLine) other;
+			if (this != fromLine) {
+				setPartialAmount(fromLine.getPartialAmount());
+				setPartialTransactionCount(fromLine.getPartialTransactionCount());
+				setStartDate(fromLine.getStartDate());
+				setEndDate(fromLine.getEndDate());
+				setScheduleType(fromLine.getScheduleType());
+				setAmount(fromLine.getAmount());
+			}
+		}
+	}
+	
+	public ScheduleTypes getScheduleTypeEnum() {
+		if (CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.DAILY.name.equals(scheduleType)) {
+			return CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.DAILY;
+		} else if(CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.WEEKLY.name.equals(scheduleType)) {
+			return CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.WEEKLY;
+		} else if(CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.MONTHLY.name.equals(scheduleType)) {
+			return CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.MONTHLY;
+		} else if(CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.YEARLY.name.equals(scheduleType)) {
+			return CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes.YEARLY;
+		} else {
+			throw new IllegalArgumentException("invalid schedule typeL: " + this.scheduleType);
```

Comment:
looks like an extra "L" after type in the message passed to the exception constructor